docs/api/primitives.md

```diff
@@ -38,6 +44,18 @@ A String indentifier that is intended to be both human and machine-readable.
 
 Decimal number.
 
+## PropertyValidator
+An identifier of any suitable validator function chosen by the implementor of service.
+
+## Protocol
+A string representation of a communication protocol.
+
+## SecurityPolicy
+Any suitable security policy chosen by the implementor of service.
+
+## String
+A chain of characters.
```

Comment:
If we specify what a string is, we should mention that only UTF-8 characters are supported.